server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/CompactionFinalizer.java

```diff
@@ -148,16 +150,22 @@ private void processPending() {
 
         List<ExternalCompactionId> statusesToDelete = new ArrayList<>();
 
+        Map<KeyExtent,TabletMetadata> tabletsMetadata = new HashMap<>();
+
+        List<KeyExtent> extents =
+            batch.stream().map(ecfs -> ecfs.getExtent()).collect(Collectors.toList());
+
+        try (TabletsMetadata tablets = context.getAmple().readTablets().forTablets(extents)
+            .fetch(ColumnType.LOCATION, ColumnType.PREV_ROW, ColumnType.ECOMP).build()) {
+          tablets.forEach(tm -> tabletsMetadata.put(tm.getExtent(), tm));
+        }
```

Comment:
Can stream directly to a map using a collector (I did a static import of `Function.identity()` and `Collectors.toMap()` to shorten the line for readability):

```suggestion
        Map<KeyExtent,TabletMetadata> tabletsMetadata;
        var extents = batch.stream().map(ExternalCompactionFinalState::getExtent).collect(toList());
        try (TabletsMetadata tablets = context.getAmple().readTablets().forTablets(extents)
            .fetch(ColumnType.LOCATION, ColumnType.PREV_ROW, ColumnType.ECOMP).build()) {
          tabletsMetadata = tablets.stream().collect(toMap(TabletMetadata::getExtent, identity()));
        }
```